cmd/agent/subcommands/flare/command.go

```diff
@@ -242,10 +242,14 @@ func readProfileData(seconds int) (flare.ProfileData, error) {
 		"security-agent": serviceProfileCollector(tcpGet("security_agent.expvar_port", false), seconds),
 	}
 
-	if pkgconfigsetup.Datadog().GetBool("process_config.enabled") ||
-		pkgconfigsetup.Datadog().GetBool("process_config.container_collection.enabled") ||
-		pkgconfigsetup.Datadog().GetBool("process_config.process_collection.enabled") {
-
+	processChecksInProcessAgent := !pkgconfigsetup.Datadog().GetBool("process_config.run_in_core_agent.enabled") &&
+		(pkgconfigsetup.Datadog().GetBool("process_config.enabled") ||
+			pkgconfigsetup.Datadog().GetBool("process_config.container_collection.enabled") ||
+			pkgconfigsetup.Datadog().GetBool("process_config.process_collection.enabled"))
+	npmEnabled := pkgconfigsetup.SystemProbe().GetBool("network_config.enabled")
+	usmEnabled := pkgconfigsetup.SystemProbe().GetBool("service_monitoring_config.enabled")
```

Comment:
Could I annoy you to use the config component rather than using the `pkgconfigsetup.Datadog()` directly 😄